### PR TITLE
Adds automatic TGS DMAPI updates

### PR DESCRIPTION
**Pull request by Cyberboss**
Runs every UTC midnight, can be run manually, makes a PR if there's an update

Sample: https://github.com/tgstation/tgstation/pull/54448

### CODE REVIEW

File: .github/workflows/update_tgs_dmapi.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Apply DMAPI update
+      uses: tgstation/tgs-dmapi-updater@v2
+      with:
+        header-path: 'code/__DEFINES/tgs.dm'
+        library-path: 'code/modules/tgs'
+
+    - name: Commit and Push
+      run: |
+        git config user.name tgstation-server
+        git config user.email tgstation-server@users.noreply.github.com
```
Comment (Crossedfall) [inline .github/workflows/update_tgs_dmapi.yml]:
```suggestion
        git config user.name "ss13-beebot"
        git config user.email "56381746+ss13-beebot@users.noreply.github.com"
```

File: .github/workflows/update_tgs_dmapi.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Create Pull Request
+      uses: repo-sync/pull-request@v2
+      with:
+        source_branch: "tgs-dmapi-update"
+        destination_branch: "master"
+        pr_title: "Automatic TGS DMAPI Update"
+        pr_body: "This pull request updates the TGS DMAPI to the latest version. Please note any breaking or unimplemented changes before merging."
+        pr_label: "Tools"
+        pr_allow_empty: false
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Crossedfall) [inline .github/workflows/update_tgs_dmapi.yml]:
```suggestion
        #github_token: ${{ secrets.GITHUB_TOKEN }} Use this instead if you have unprotected branches
        github_token: ${{ secrets.CL_TOKEN }}
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Mighty kind of you! Just a few changes needed to match up with our other actions.

File: .github/workflows/update_tgs_dmapi.yml (hunk 2 above)
Comment (Cyberboss) [reply .github/workflows/update_tgs_dmapi.yml]: This shouldn't be needed though, unless you have a wildcard for your branch protection? It creates and pushes to the branch `tgs-dmapi-update`.

Comment (Crossedfall) [reply .github/workflows/update_tgs_dmapi.yml]: Was doing it mostly for consistency sake. You are correct that it's not strictly required, however.

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**